src/simplify_algebra.cpp

```diff
@@ -851,6 +851,31 @@ struct find_div_const
     }
 };
 
+struct find_unit_mult_const
+{
+    auto matcher() const
+    {
+        return match::name("mul")(
+            match::either_arg(0, 1)(match::has_value(1.0f), match::any().bind("x")));
+    }
+
+    void apply(module& m, const match::matcher_result& r) const
+    {
+        auto ins  = r.result;
+        auto args = ins->inputs();
+
+        auto c_in = r.instructions["x"];
+
+        auto res = args.front();
```

Comment:
I don't think you need to check and replace for the order for `x`. `x` is bound to the non-identity argument to `ins`. You should be able to just replace `ins` with `c_in`.

Reply:
Oh thanks for pointing that out! Just rebuilt and tried that and you're correct since the test case I added checked for commutativity. So for future searches, because I'm using either_arg(0,1) the bind will always match the "other" thing then?

Reply:
Yes, bind will match to instruction that is not `1`